public/js/index.js

```diff
@@ -5,13 +5,14 @@ var defaultExtraKeys = {
       cm.setOption("fullScreen", !cm.getOption("fullScreen"));
     },
     "Esc": function(cm) {
-      if (cm.getOption("fullScreen")) cm.setOption("fullScreen", false);
+       if (cm.getOption('keyMap').substr(0, 3) === 'vim') return CodeMirror.Pass;
```

Comment:
I think you can do a complete compare here since no other input mode is similar to `vim` like `cm.getOption('keyMap') === 'vim'`